File: src/completed/java/euler0028.java
/**
 * Created by Holden on 12/19/2016. - COMPLETED
 *
 * Problem 28 - Number Spiral Diagonals
 *
 */

public class Prob0028 {
    public static void main(String[] args) {
        long startTime = System.nanoTime();
        int answer = quadraticPrimes(1001);
        long endTime = System.nanoTime();

        System.out.println(answer);

        long duration = (endTime - startTime);
        EulerTime.time(duration);
    }

    public static int quadraticPrimes(int dimension) {
        int sum = 1;
        int currentInteger = 1;
        int currentDimension = 1;

        while (currentDimension < dimension) {
            for (int i = 0; i < 4; i++) {
                currentInteger += (currentDimension + 1);
                sum += currentInteger;
            }

            currentDimension += 2;
        }

        return sum;
    }

}

File: src/completed/python/euler0028.py
###
#
# Created by Holden on 6/12/2016
#
# SOLVED on 6/12/2016
#
# Problem 28 - Number Spiral Diagonals
#
###

def number_spiral_diagonals(diameter):
    radius = diameter // 2
    total = 1
    current = 1
    
    for i in range(1, radius + 1):
        for j in range(4):
            current += 2*i
            total += current
    
    return total